file_management/src/file_upload/file_manager_strategy.cpp

```diff
@@ -273,8 +273,12 @@ bool FileManagerStrategy::isDataAvailable() {
   return !active_read_file_.empty() || !stored_files_.empty() || active_write_file_size_ > 0;
 }
 
-bool FileManagerStrategy::discardOldLogs() {
-  return options_.discard_old_logs;
+bool FileManagerStrategy::isDeleteStaleData() {
+  return options_.delete_stale_data;
+}
+
+void FileManagerStrategy::setDeleteStaleData(bool set_stale_data) {
```

Comment:
Do we really need a setter? All of the other options are set through the constructor and not (if memory serves correct) modified thereafter: why would we need to modify this option after construction?